src/passes/OptimizeInstructions.cpp

```diff
@@ -395,6 +395,24 @@ struct OptimizeInstructions
           return replaceCurrent(un);
         }
       }
+      {
+        // x < 0 || x > POSITIVE_CONST   ==>   unsigned(x) > POSITIVE_CONST
+        Binary* bin;
+        Expression *xLeft, *xRight;
+        Const* y;
+        if (matches(curr,
+                    binary(&bin,
+                           OrInt32,
+                           binary(LtSInt32, any(&xLeft), i32(0)),
+                           binary(GtSInt32, any(&xRight), constant(&y)))) &&
+            y->type == Type::i32 && y->value.geti32() > 0 &&
```

Comment:
```suggestion
            y->value.geti32() > 0 &&
```
The type must be `i32` because the operation is `GtSInt32`.